pkg/watcher/certificate_manager.go

```diff
@@ -116,33 +116,39 @@ func (c *CertificateManager) Remove(ctx context.Context) error {
 
 func (c *CertificateManager) RemoveCertificate(ctx context.Context) error {
 	certificate := &certmanagerv1.Certificate{}
-	if err := c.kcpClient.Get(ctx, client.ObjectKey{
+	err := c.kcpClient.Get(ctx, client.ObjectKey{
 		Name:      c.certificateName,
 		Namespace: c.config.IstioNamespace,
-	}, certificate); err != nil && !util.IsNotFound(err) {
+	}, certificate)
+	if err != nil && !util.IsNotFound(err) {
 		return fmt.Errorf("failed to get certificate: %w", err)
 	}
 
-	if err := c.kcpClient.Delete(ctx, certificate); err != nil {
-		return fmt.Errorf("failed to delete certificate: %w", err)
+	if !util.IsNotFound(err) {
```

Comment:
you already return error above when `!util.IsNotFound(err) `, is it possible to enter here?

Reply:
The function continues execution if the error is not nil and the !util.IsNotFound(err), so we don't want to do the delete call if the error is not found. Or I can have the check as `if err == nil`

Reply:
yes, then you should not use `!util.IsNotFound(err)`, that's misleading.